types/index.d.ts

```diff
@@ -12,8 +12,12 @@ type NoLiteralType<T> =
 
 declare namespace math {
   type MathNumericType = number | BigNumber | Fraction | Complex;
+
+  // TODO Make the inner numeric type generic
   type MathArray = MathNumericType[] | MathNumericType[][];
+  // TODO Make the inner numeric type generic
   type MathCollection = MathArray | Matrix;
+
```

Comment:
I'll probably open a future PR to properly type these collection types at some point